drivers/stepper/stepper_context.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020 Teslabs Engineering S.L.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_STEPPER_STEPPER_CONTEXT_H_
+#define ZEPHYR_DRIVERS_STEPPER_STEPPER_CONTEXT_H_
+
+#include <kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct stepper_context {
```

Comment:
This `foo_context` header replicates a pattern that we know is wrong.  It supports asynchronous by adding a blocking async method, which is confusing.  It also uses a semaphore for coordination while a mutex may be more appropriate (not withstanding #25678).

I don't know that there's an alternative right now, but it might be wise to not provide an async API until we have a newer interface that addresses these concerns.